plugins/utils/nvim-colorizer.nix

```diff
@@ -0,0 +1,166 @@
+      default = null;
+    };
+    names = mkOption {
+      description = "\"Name\" codes like Blue or blue";
+      type = types.nullOr types.bool;
+      default = null;
+    };
+    RRGGBBAA = mkOption {
+      description = "#RRGGBBAA hex codes";
+      type = types.nullOr types.bool;
+      default = null;
+    };
+    AARRGGBB = mkOption {
+      description = "0xAARRGGBB hex codes";
+      type = types.nullOr types.bool;
+      default = null;
+    };
+    rgb_fn = mkOption {
+      description = "CSS rgb() and rgba() functions";
+      type = types.nullOr types.bool;
+      default = null;
+    };
+    hsl_fn = mkOption {
+      description = "CSS hsl() and hsla() functions";
+      type = types.nullOr types.bool;
+      default = null;
+    };
+    css = mkOption {
+      description = "Enable all CSS features: rgb_fn, hsl_fn, names, RGB, RRGGBB";
+      type = types.nullOr types.bool;
+      default = null;
+    };
+    css_fn = mkOption {
+      description = "Enable all CSS *functions*: rgb_fn, hsl_fn";
+      type = types.nullOr types.bool;
+      default = null;
+    };
+    mode = mkOption {
+      description = "Set the display mode";
+      type = types.nullOr (types.enum [ "foreground" "background" "virtualtext" ]);
+      default = null;
+    };
+    tailwind = mkOption {
+      description = "Enable tailwind colors";
+      type = types.nullOr (
+        types.oneOf [
+          types.bool
+          (types.enum [ "normal" "lsp" "both" ])
+        ]
+      );
+      default = null;
+    };
+    sass = mkOption {
+      description = "Enable sass colors";
+      type = types.nullOr (
+        types.submodule {
+          options = {
+            enabled = mkOption {
+              type = types.nullOr types.bool;
+              default = null;
+            };
+            parsers = mkOption {
+              type = types.nullOr types.attrs;
+              default = null;
+            };
+          };
+        }
+      );
+      default = null;
+    };
+    virtualtext = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+    };
+  };
+
+in
+{
+  options = {
+    plugins.nvim-colorizer = {
+
+      enable = mkEnableOption "nvim-colorizer";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.vimPlugins.nvim-colorizer-lua;
+        description = "Plugin to use for vim-commentary";
+      };
+
+      fileTypes = mkOption {
+        description = "Enable and/or configure highlighting for certain filetypes";
+        type = types.nullOr (
+          types.listOf (types.oneOf [
+            types.str
+            (types.submodule {
+              options = {
+                language = mkOption {
+                  type = types.str;
+                };
+              } // colorizer-options;
+            })
+          ])
+        );
+        default = null;
+      };
+
+      userDefaultOptions = mkOption {
+        description = "Defaults options";
```

Comment:
Think there's a typo here, shouldn't it be "Default options"?

Reply:
Indeed, fixed.